opennlp-tools/src/main/java/opennlp/tools/util/normalizer/UrlCharSequenceNormalizer.java

```diff
@@ -24,9 +24,9 @@
 public class UrlCharSequenceNormalizer implements CharSequenceNormalizer {
 
   private static final Pattern URL_REGEX =
-      Pattern.compile("https?://[-_.?&~;+=/#0-9A-Za-z]+");
+      Pattern.compile("https?://[-_.?&~;+=/#0-9A-Za-z]{1,10000}");
   private static final Pattern MAIL_REGEX =
-      Pattern.compile("[-_.0-9A-Za-z]+@[-_0-9A-Za-z]+[-_.0-9A-Za-z]+");
+      Pattern.compile("[-_.0-9A-Za-z]{1,100}@[-_0-9A-Za-z]{1,100}[-_.0-9A-Za-z]{1,100}");
```

Comment:
Are these limits are limiting the length of the URL that can be detected? And why is it important to add it?

Reply:
Y.  These do limit the lengths.  This was one solution that prevented worse than linear processing times on long strings of text.  When Tika updated its regression corpus, one doc had a really long string of DNA with no spaces.  This caused a similar regex in Optimaize to take > 5 minutes to process.

See the stats here: https://issues.apache.org/jira/browse/OPENNLP-1266?focusedCommentId=16858605&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16858605

![image](https://user-images.githubusercontent.com/6739646/59698103-00d02c00-91bd-11e9-8a78-5a5e823d6ad5.png)